### DISCUSSION TITLE
Generate dynamic form based on JSON DSL

### DISCUSSION THREAD

**Discussion by fbdajaysharma** [adobe/react-spectrum#4857] [Ideas]
Hi All,

My team mate (@deepchand) and I are working on a feature which can help generate dynamic forms based on JSON DSL.
It will help teams to generate forms by just specifying the DSL.
This is useful when there are scenarios where basic form structure don't change but teams wants form to be dynamic based on different requirements.

For example:- in our use case there could be multiple jobs which require diff input type, teams requests these jobs by just specifying the JSON DSL and don't write the code for UI.

I was wondering if other teams have similar requirement. Our goal is to open source this development and provide it as component under react spectrum

**Comment by deepchand** [comment]
For more information have a look on below JSON DSL which will render a Dynamic Form like below
```
{
    "name": {
        "type": "string",
        "label": "Name",
        "defaultValue": "Deep"
    },
    "age": {
        "type": "integer",
        "label": "Age",
        "defaultValue": 21
    },
    "tnc": {
        "type": "boolean",
        "label": "Terms and Conditions",
        "defaultValue": true
    },
    "city": {
        "type": "dropdown",
        "options": [
            {
                "value": "delhi",
                "label": "New Delhi"
            },
            {
                "value": "noida",
                "label": "Noida"
            },
            {
                "value": "gurgaon",
                "label": "Gurgaon"
            }
        ],
        "label": "City",
        "defaultValue": "noida"
    }
}

```
<img width="390" alt="Screenshot 2023-08-03 at 9 37 48 AM" src="https://github.com/adobe/react-spectrum/assets/18530606/0f1708c5-84ba-412c-a9bd-b9cc27f84248">
